mopro-ffi/src/circom/mod.rs

```diff
@@ -142,33 +145,23 @@ pub fn generate_circom_proof_wtns(
     // this loader will only load the first few bytes
     let mut header_reader = ZkeyHeaderReader::new(&zkey_path);
     header_reader.read();
-    let file = File::open(&zkey_path).map_err(|e| MoproError::CircomError(e.to_string()))?;
+    let file = File::open(&zkey_path)?;
     let mut reader = std::io::BufReader::new(file);
     // check the prime in the header
     // println!("{} {} {}", header.q, header.n8q, ark_bls12_381::Fq::MODULUS);
     if header_reader.r == BigUint::from(ark_bn254::Fr::MODULUS) {
-        let (proving_key, matrices) = read_zkey::<_, Bn254>(&mut reader)
-            .map_err(|e| MoproError::CircomError(e.to_string()))?;
-        let full_assignment = witness_thread
-            .join()
-            .map_err(|_e| MoproError::CircomError("Failed to generate witness".to_string()))?;
+        let (proving_key, matrices) = read_zkey::<_, Bn254>(&mut reader)?;
+        let full_assignment = witness_thread.join().unwrap();
```

Comment:
will the error messages here be caught or shown? if there are errors.
e.g. previously there are errors `e.to_string()` or `"Failed to generate witness"`

Reply:
It would cause a panic. I caught it and provided a nicer error message.